### ISSUE TITLE
macOS-architecture RIDs

### ISSUE THREAD

**Issue by TheFanatr** [dotnet/runtime#44931]
> By the time .NET 6 is released, it will have been more than five years since Apple renamed this operating system to "macOS". It's not called "OS X" any more. If it can be done correctly in TFMs, it can be done correctly in RIDs.

_Originally posted by @warrenrumak in https://github.com/dotnet/runtime/issues/43313#issuecomment-707261960_

This is in reference to the fact that RIDs for macOS are still `osx-architecture` (and seems like it will be `osx-arm64` for Apple Silicon Macs); this should be fixed at this point before it becomes more set in stone.

@richlander replied to the above quoted comment, for the OP to file an issue, but they haven't and it's been a month, so here is mine.

Hope this gets done; if there's already an issue for this or it was already done, I didn't find it or notice, and someone can close.

**Comment by danmosemsft** [comment]
@terrajobst ?

**Comment by am11** [comment]
<kbd>\* NO MERGE \*</kbd>
but this is not a PR. 🙂

**Comment by aaronfranke** [comment]
@danmosemsft This issue is labeled `* NO MERGE *` but in #43313 it says "Support compilation targeting osx-arm64" and the box is checked. So this has already been merged, and should be fixed soon before it's too late.

**Comment by danmosemsft** [comment]
Thanks for pointing that out. As much as it's not ideal as is, I expect doing this would require cutting work in other places including NuGet. Could you say a little more about what you believe the relative priority of this should be and why?

Cc @terrajobst @richlander @wfurt @sdmaclea

**Comment by aaronfranke** [comment]
@danmosemsft The issue is that the longer we wait the bigger the potential breakage is. Since .NET 6 introduces new names (the mentioned inaccurate `osx-arm64`), the optimal strategy is to use the correct names from the start, and then the only breakages that have to be worried about are with the existing code. So I think this should be solved before .NET 6 is in preview; if .NET 6 follows [a similar schedule to .NET 5](https://devblogs.microsoft.com/dotnet/announcing-net-5-0-preview-1/), this means that this issue should be solved within the next few months.

**Comment by danmosemsft** [comment]
Cc @ericstj @marek-safar

**Comment by sdmaclea** [comment]
We will probably need to create `macos` & `macos-*` as an alias for `osx` & `osx-*`.  I am not sure `osx-arm64` makes the situation any worse as we would probably want to keep things symmetrical.

**Comment by TheFanatr** [comment]
I think it's also somewhat useful to point out that not only is the operating system no longer called Mac OS X, macOS Big Sur is actually version 11 of the software, as opposed to 10 for the multiple previous macOS releases, so it's incorrect on multiple levels.

**Comment by marek-safar** [comment]
> like it will be osx-arm64 for Apple Silicon Macs

yep, that's correct, and [current setup](https://github.com/wfurt/runtime/blob/master/src/libraries/pkg/Microsoft.NETCore.Platforms/runtimeGroups.props#L121-L125)

I agree there should be a way to alias OSX to macOS and vice-versa so developers can use both names on CLI and in their projects/NuGets. I don't think we should be pedantic and block osx-arm64 though.

**Comment by ericstj** [comment]
Looks like this issue might be tracking the same thing as https://github.com/dotnet/runtime/issues/44120

**Comment by krwq** [comment]
I assume this is now out of scope for 6.0?

**Comment by krwq** [comment]
Changed to future since the other issue is marked as Future
